### ISSUE TITLE
response doesn't go through middlewares

### ISSUE THREAD

**Issue by bunnyc1986** [Exilz/react-native-offline-api#9]
Is it possible to add ability to process response through middlewares? 
In many cases, I need to handle http status code or reformat the response body, and I'd like to write a middleware for these purposes.

**Comment by Exilz** [comment]
Hi,
this is an interesting feature.

I'll think about the best way to implement this and get back to you.

**Comment by bunnyc1986** [comment]
thanks. I look forward to hearing from you.

**Comment by Exilz** [comment]
Hi,
you can try the new `responseMiddleware` option in the 2.3.0 PR #15

**Comment by treemore** [comment]
That's great. Twenty minutes before I discovered this library, I really wanted it. Still considering how to deal with jwt expired. This feature is exactly what we need.
There is a question,
https://github.com/Exilz/react-native-offline-api/blob/dd8e0d82124614000beba311e80d83c391d9ba57/docs/middlewares.md
Here res can get response headers and statusCode? The body returned by the backend is in res.body?

**Comment by treemore** [comment]
or the res is just the object return by the fetch api ? 
Forgive me for not doing my own experiment, but I think it would be better to understand if the example shows more details.

**Comment by Exilz** [comment]
That's some great timing then 😄 

The `res` argument in the example you linked is the body, the headers aren't accessible there.

When I wrote this, I thought that headers shouldn't have to be modified anyway, but it's still open to debate.

Do you think it would be useful to be able to alter both the body and the headers ?

**Comment by treemore** [comment]
Yes. I think statusCode and headers are important. I will not change these things in my needs, but I may read them to make decisions. For example, once the statusCode=401 returned by the backend is found, I think the token has timed out and then refresh the token.

**Comment by Exilz** [comment]
You make a point there, I'll change the code to make sure that's possible.

However, I'm not sure how your token example applies to this `responseMiddleware` option. Since you need to update your token, I guess you have to fire another network request anyway, so how does this help you ?

I'm not really familiar with jwt, I might be missing something here.

**Comment by treemore** [comment]
Not necessarily jwt. Just like your example here.
https://github.com/Exilz/react-native-offline-api/blob/master/docs/middlewares.md.
For example, the token's expiration time is 10 minutes.
Ten minutes expired after 10 minutes. The next request will be abnormal. If there is not a single place to deal with the logic here, then every place that uses the service must add a duplicate processing logic. Although I can deal with this problem by adding a layer above your framework, if you can open the original response object can save us a lot of time.
